eclass/xemacs-packages-r1.eclass

```diff
@@ -0,0 +1,99 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+# @ECLASS: xemacs-packages-r1.eclass
+# @MAINTAINER:
+# xemacs@gentoo.org
+# @BLURB: Eclass to support elisp packages distributed by XEmacs.
+# @DESCRIPTION:
+# This eclass supports ebuilds for packages distributed by XEmacs.
+# Ebuilds have to support at least EAPI 6.
+
+case ${EAPI:-0} in
+	[0-5])
+		die "xemacs-packages-r1.eclass is banned in EAPI ${EAPI:-0}"
+		;;
+	6)
+		;;
+	*)
+		die "Unknown EAPI ${EAPI:-0}"
+		;;
+esac
+
+if [[ ! ${_XEMACS_PACKAGES_R1} ]]; then
+
+EXPORT_FUNCTIONS src_install
```

Comment:
Better keep EXPORT_FUNCTIONS outta guard, so that the order is respected when the eclass is inherited twice. Not that it will happen but still.